lib/src/configuration.cc

```diff
@@ -683,6 +683,22 @@ void Configuration::validateAndNormalizeOtherSettings() {
         }
 
         auto pid_dir = fs::path(pid_file).parent_path().string();
+
+#ifdef __sun
+        // Solaris cleans up /var/run when rebooting; daemonization
+        // assumes that the PID file directory exists. If we're using
+        // the default directory, make sure it exists (PCP-297)
+        if (pid_dir == "/var/run/puppetlabs" && !fs::exists(pid_dir)) {
```

Comment:
Should we create the directory if it doesn't exist even if it's not the default directory? The counter-argument to that idea would be that if someone has specified the pidfile manually but typo'd the path, they wouldn't want the directory created.

If we do restrict this logic to the default directory, can we DRY out occurences of `/var/run/puppetlabs` and use `DEFAULT_PID_FILE` above?